python/ray/data/impl/shuffle.py

```diff
@@ -103,13 +104,24 @@ def simple_shuffle(
 
 def _shuffle_map(
     block: Block,
+    block_udf: Optional[Callable[[Block], Iterable[Block]]],
     idx: int,
     output_num_blocks: int,
     random_shuffle: bool,
     random_seed: Optional[int],
 ) -> List[Union[BlockMetadata, Block]]:
     """Returns list of [BlockMetadata, O1, O2, O3, ...output_num_blocks]."""
     stats = BlockExecStats.builder()
+    if block_udf:
+        # TODO(ekl) note that this effectively disables block splitting.
+        pieces = list(block_udf(block))
+        if len(pieces) > 1:
+            builder = BlockAccessor.for_block(pieces[0]).builder()
+            for p in pieces:
+                builder.add_block(p)
+            block = builder.build()
+        else:
+            block = pieces[0]
```

Comment:
Nit: Found the `pieces` naming to be odd.
```suggestion
        blocks = list(block_udf(block))
        if len(blocks) > 1:
            builder = BlockAccessor.for_block(blocks[0]).builder()
            for b in blocks:
                builder.add_block(b)
            block = builder.build()
        else:
            block = blocks[0]
```

Reply:
Thanks. I'll fix this in the follow-up